lib/libc/syslog/lib_syslog.c

```diff
@@ -156,7 +156,7 @@ static inline int vsyslog_internal(FAR const char *fmt, va_list ap)
 	 * do the work.
 	 */
 
-	lib_rawoutstream(&stream, 1);
+	lib_stdoutstream(&stream, stdout);
 
 #if CONFIG_NFILE_STREAMS > 0 && (defined(CONFIG_BUILD_FLAT) || defined(__KERNEL__))
 	if(!up_interrupt_context() && getpid() != 0){
```

Comment:
if we use lib_stdoutstream we don't need to use lib_take_semaphore(stdout) here

Reply:
without using lib_take_semaphore , we will again get log_mix up . It is necessary, so that the complete string gets printed without mixup

Reply:
oh you are right !!
it is needed.